### PR TITLE
SP: strip unlearned columns: removed

### PR DESCRIPTION
**Pull request by breznak**
removed strip

Fixes: #281

### CODE REVIEW

File: src/nupic/algorithms/SpatialPooler.cpp [hunk 1]
```diff
@@ -524,26 +524,31 @@ void SpatialPooler::compute(SDR &input, bool learn, SDR &active) {
...
+  size_t removed = active.size;
   for (const auto& col: sparseCols) {
     if (activeDutyCycles_[col] > 0) {
       res.push_back(col);
     }
   }
   //update original SDR with changed values
   active.setFlatSparse(res);
+  removed = removed - active.size; //size of active befor/after the pruning
+  if(removed > 0) std::cout << "stripped: " << removed << std::endl;
```
Comment (breznak) [inline src/nupic/algorithms/SpatialPooler.cpp]: print if any cols are removed by this method, actually.

**Comment by breznak** [comment]
Happy saint David's day, @dkeeney @ctrl-z-9000-times @cogmission @david-ragazzi ! :tada: :balloon: :clinking_glasses:  Thanks for your ideas and work!!

**Comment by ctrl-z-9000-times** [comment]
Thank you Breznak for doing this.  This is one less thing for users to worry about!

File: API_CHANGELOG.md [hunk 2]
```diff
@@ -57,4 +57,4 @@ are ignored.  PR #271
 
 * Removed all matrix libraries.  Use the `Connections` class instead.  PR #169
 
-* Removed `void SpatialPooler::stripUnlearnedColumns()` as unused and not useful (did not effectively remove any columns). https://github.com/htm-community/nupic.cpp/pull/286 
+* Removed `void SpatialPooler::stripUnlearnedColumns()` as unused and not useful (did not effectively remove any columns). PR #286
```
Comment (breznak) [inline API_CHANGELOG.md]: @ctrl-z-9000-times just a mention, next time I'd prefer the full URL format. To a user reading the document in browser, it both shows as "#286", but the `#286` breaks if you copy the code to a repo, or what not.

Comment (ctrl-z-9000-times) [reply API_CHANGELOG.md]: If that is the case then you should update all of the cross references in the API_Changelog.  I changed it to a short link for consistency with the rest of the file.

Comment (breznak) [reply API_CHANGELOG.md]: > update all of the cross references in the API_Changelog

well, I probably don;t care enough to do that now. Was just saying for next time...

**Pull request merged**